test_obc_creation_and_deletion fails to find created OBC. OBC created in wrong project

https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#ocs/launches/557/13901/631968/632117/632120/log

E               ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: oc --kubeconfig /home/jenkins/current-cluster-dir/openshift-cluster-dir/auth/kubeconfig -n openshift-storage get obc obc-ui-7f667b280f714326a72bb4db25d09307 -n openshift-storage -o yaml.
E               Error is Error from server (NotFound): objectbucketclaims.objectbucket.io "obc-ui-7f667b280f714326a72bb4db25d09307" not found

-----------
OBC got created in wrong project even though test states "openshift-storage" has been selected

NAMESPACE                            NAME                                      BUCKET-NAME                                                       STORAGE-CLASS                 BUCKET-CLASS                  PHASE   
openshift-cluster-samples-operator   obc-ui-7f667b280f714326a72bb4db25d09307   obc-ui-7f667b280f714326a72-6614cb70-5342-4ef1-827a-63ee0f37c8bc   openshift-storage.noobaa.io   noobaa-default-bucket-class   Bound   

-----------
We need to add screenshot and UI validation that we indeed on openshift-storage page.

--- a/ocs_ci/ocs/ui/base_ui.py
+++ b/ocs_ci/ocs/ui/base_ui.py
@@ -507,33 +507,19 @@ class BaseUI:
         element = self.driver.find_element(locator[1], locator[0])
         actions.move_to_element(element).perform()
 
-    def take_screenshot(self):
+    def take_screenshot(self, name_suffix: str = ""):
         """
         Take screenshot using python code
 
         """
-        time.sleep(1)
-        filename = os.path.join(
-            self.screenshots_folder,
-            f"{datetime.datetime.now().strftime('%Y-%m-%dT%H-%M-%S.%f')}.png",
-        )
-        logger.debug(f"Creating screenshot: {filename}")
-        self.driver.save_screenshot(filename)
-        time.sleep(0.5)
+        take_screenshot(self.screenshots_folder, name_suffix)
 
-    def copy_dom(self):
+    def copy_dom(self, name_suffix: str = ""):
         """
         Get page source of the webpage
 
         """
-        filename = os.path.join(
-            self.dom_folder,
-            f"{datetime.datetime.now().strftime('%Y-%m-%dT%H-%M-%S.%f')}_DOM.txt",
-        )
-        logger.info(f"Copy DOM file: {filename}")
-        html = self.driver.page_source
-        with open(filename, "w") as f:
-            f.write(html)
+        copy_dom(self.dom_folder, name_suffix)
 
     def do_clear(self, locator, timeout=30):
         """
@@ -656,14 +642,16 @@ def screenshot_dom_location(type_loc="screenshot"):
         )
 
 
-def copy_dom(name_suffix: str = ""):
+def copy_dom(dom_folder=None, name_suffix: str = ""):
     """
     Copy DOM using python code
 
     Args:
+        dom_folder (str): path to folder where dom text file will be saved
         name_suffix (str): name suffix, will be added before extension. Optional argument
     """
-    dom_folder = screenshot_dom_location(type_loc="dom")
+    if dom_folder is None:
+        dom_folder = screenshot_dom_location(type_loc="dom")
     if not os.path.isdir(dom_folder):
         Path(dom_folder).mkdir(parents=True, exist_ok=True)
     time.sleep(1)
@@ -680,14 +668,16 @@ def copy_dom(name_suffix: str = ""):
     time.sleep(0.5)
 
 
-def take_screenshot(name_suffix: str = ""):
+def take_screenshot(screenshots_folder=None, name_suffix: str = ""):
     """
     Take screenshot using python code
 
     Args:
+        screenshots_folder (str): path to folder where screenshot will be saved
         name_suffix (str): name suffix, will be added before extension. Optional argument
     """
-    screenshots_folder = screenshot_dom_location(type_loc="screenshot")
+    if screenshots_folder is None:
+        screenshots_folder = screenshot_dom_location(type_loc="screenshot")
     if not os.path.isdir(screenshots_folder):
         Path(screenshots_folder).mkdir(parents=True, exist_ok=True)
     time.sleep(1)

--- a/ocs_ci/ocs/ui/base_ui.py
+++ b/ocs_ci/ocs/ui/base_ui.py
@@ -13,6 +13,7 @@ from selenium.common.exceptions import (
     WebDriverException,
     NoSuchElementException,
     StaleElementReferenceException,
+    ElementClickInterceptedException,
 )
 from selenium.webdriver.remote.webdriver import WebDriver
 from selenium.webdriver.chrome.options import Options
@@ -217,6 +218,12 @@ class BaseUI:
                 _do_click(locator, timeout, enable_screenshot, copy_dom)
             else:
                 raise
+        except ElementClickInterceptedException:
+            # appears due to JS graphics on the page: one element overlapping another, or dynamic graphics in progress
+            logger.info("ElementClickInterceptedException, try click again")
+            take_screenshot("ElementClickInterceptedException")
+            time.sleep(5)
+            _do_click(locator, timeout, enable_screenshot, copy_dom)
 
     def do_click_by_id(self, id, timeout=30):
         return self.do_click((id, By.ID), timeout)

--- a/ocs_ci/ocs/ui/base_ui.py
+++ b/ocs_ci/ocs/ui/base_ui.py
@@ -512,14 +512,16 @@ class BaseUI:
         Take screenshot using python code
 
         """
-        take_screenshot(self.screenshots_folder, name_suffix)
+        take_screenshot(
+            screenshots_folder=self.screenshots_folder, name_suffix=name_suffix
+        )
 
     def copy_dom(self, name_suffix: str = ""):
         """
         Get page source of the webpage
 
         """
-        copy_dom(self.dom_folder, name_suffix)
+        copy_dom(dom_folder=self.dom_folder, name_suffix=name_suffix)
 
     def do_clear(self, locator, timeout=30):
         """
@@ -642,13 +644,13 @@ def screenshot_dom_location(type_loc="screenshot"):
         )
 
 
-def copy_dom(dom_folder=None, name_suffix: str = ""):
+def copy_dom(name_suffix: str = "", dom_folder=None):
     """
     Copy DOM using python code
 
     Args:
-        dom_folder (str): path to folder where dom text file will be saved
         name_suffix (str): name suffix, will be added before extension. Optional argument
+        dom_folder (str): path to folder where dom text file will be saved
     """
     if dom_folder is None:
         dom_folder = screenshot_dom_location(type_loc="dom")
@@ -668,13 +670,13 @@ def copy_dom(dom_folder=None, name_suffix: str = ""):
     time.sleep(0.5)
 
 
-def take_screenshot(screenshots_folder=None, name_suffix: str = ""):
+def take_screenshot(name_suffix: str = "", screenshots_folder=None):
     """
     Take screenshot using python code
 
     Args:
-        screenshots_folder (str): path to folder where screenshot will be saved
         name_suffix (str): name suffix, will be added before extension. Optional argument
+        screenshots_folder (str): path to folder where screenshot will be saved
     """
     if screenshots_folder is None:
         screenshots_folder = screenshot_dom_location(type_loc="screenshot")

--- a/ocs_ci/ocs/ui/page_objects/page_navigator.py
+++ b/ocs_ci/ocs/ui/page_objects/page_navigator.py
@@ -1,15 +1,12 @@
-from selenium.common.exceptions import NoSuchElementException
-from selenium.webdriver.support import expected_conditions as ec
-from selenium.webdriver.support.wait import WebDriverWait
-
+from selenium.common.exceptions import (
+    NoSuchElementException,
+)
 from ocs_ci.framework import config
 from ocs_ci.ocs import constants
-from ocs_ci.ocs.exceptions import TimeoutExpiredError
 from ocs_ci.ocs.ocp import get_ocp_url
 from ocs_ci.ocs.ui.base_ui import BaseUI, logger
 from ocs_ci.ocs.ui.views import ODF_OPERATOR, OCS_OPERATOR
 from ocs_ci.utility import version
-from ocs_ci.utility.utils import TimeoutSampler
 
 
 class PageNavigator(BaseUI):
@@ -366,39 +363,35 @@ class PageNavigator(BaseUI):
 
         from ocs_ci.ocs.ui.helpers_ui import format_locator
 
-        if self.ocp_version_full in (version.VERSION_4_10, version.VERSION_4_11):
-            default_projects_is_checked = self.driver.find_element_by_xpath(
-                "//span[@class='pf-c-switch__toggle']"
-            )
+        # if project is already selected, skip and proceed further
+        if self.get_elements(
+            format_locator(self.generic_locators["project_selected"], project_name)
+        ):
+            self.take_screenshot("namespace_selected")
+            logger.info("Project already selected")
+            return True
 
-            if (
-                default_projects_is_checked.get_attribute("data-checked-state")
-                == "false"
-            ):
-                logger.info("Show default projects")
-                self.do_click(self.page_nav["show-default-projects"])
-        else:
-            default_projects_is_checked = self.driver.find_element_by_css_selector(
-                "input[class='pf-c-switch__input']"
-            )
-            if (
-                default_projects_is_checked.get_attribute("data-checked-state")
-                == "false"
-            ):
-                logger.info("Show default projects")
-                self.do_click(self.page_nav["show-default-projects"])
+        default_projects_is_checked = self.driver.find_element_by_css_selector(
+            "input[class='pf-c-switch__input']"
+        )
+        if default_projects_is_checked.get_attribute("data-checked-state") == "false":
+            logger.info("Show default projects")
+            self.do_click(self.page_nav["show-default-projects"])
 
         logger.info(f"Wait and select namespace {project_name}")
         wait_for_project = self.wait_until_expected_text_is_found(
-            locator=format_locator(self.pvc_loc["test-project-link"], project_name),
+            locator=format_locator(
+                self.generic_locators["test-project-link"], project_name
+            ),
             expected_text=f"{project_name}",
             timeout=10,
         )
         if wait_for_project:
             self.do_click(
-                format_locator(self.pvc_loc["test-project-link"], project_name)
+                format_locator(self.generic_locators["test-project-link"], project_name)
             )
             logger.info(f"Namespace {project_name} selected")
+            return True
         else:
             raise NoSuchElementException(f"Namespace {project_name} not found on UI")
 
@@ -414,28 +407,9 @@ class PageNavigator(BaseUI):
             bool: True if the resource was found, False otherwise
         """
 
-        def _retrieve_current_status_from_ui():
-            resource_status = WebDriverWait(self.driver, timeout).until(
-                ec.visibility_of_element_located(
-                    self.generic_locators["resource_status"][::-1]
-                )
-            )
-            logger.info(f"Resource status is {resource_status.text}")
-            return resource_status
-
         logger.info(
             f"Verifying that the resource has reached a {status_to_check} status"
         )
-        try:
-            for resource_ui_status in TimeoutSampler(
-                timeout,
-                3,
-                _retrieve_current_status_from_ui,
-            ):
-                if resource_ui_status.text.lower() == status_to_check.lower():
-                    return True
-        except TimeoutExpiredError:
-            logger.error(
-                "The resource did not reach the expected state within the time limit."
-            )
-            return False
+        return self.wait_until_expected_text_is_found(
+            self.generic_locators["resource_status"], status_to_check, timeout
+        )

--- a/ocs_ci/ocs/ui/page_objects/page_navigator.py
+++ b/ocs_ci/ocs/ui/page_objects/page_navigator.py
@@ -369,6 +369,9 @@ class PageNavigator(BaseUI):
         ):
             self.take_screenshot("namespace_selected")
             logger.info("Project already selected")
+            self.do_click(
+                format_locator(self.generic_locators["project_selected"], project_name)
+            )
             return True
 
         default_projects_is_checked = self.driver.find_element_by_css_selector(

--- a/ocs_ci/ocs/ui/views.py
+++ b/ocs_ci/ocs/ui/views.py
@@ -259,6 +259,16 @@ generic_locators = {
         By.CSS_SELECTOR,
     ),
     "close_modal_btn": ("//button[@id='modal-close-action']", By.XPATH),
+    # project name in the dropdown header
+    "project_selected": (
+        "//span[@class='pf-c-menu-toggle__text' and contains(text(), 'Project: {}')]",
+        By.XPATH,
+    ),
+    # project name in the dropdown list
+    "test-project-link": (
+        "//li[@class='pf-c-menu__list-item']/descendant::*//*[contains(text(), '{}')]",
+        By.XPATH,
+    ),
 }
 
 ocs_operator_locators = {
@@ -497,7 +507,11 @@ pvc_4_9 = {
 }
 
 pvc_4_10 = {
-    "test-project-link": ("(//*[contains(text(),'{}')])[1]", By.XPATH),
+    # similar to generic["test-project-link"]
+    "test-project-link": (
+        "//li[@class='pf-c-menu__list-item']/descendant::*//*[contains(text(), '{}')]",
+        By.XPATH,
+    ),
 }
 
 pvc_4_12 = {
@@ -970,10 +984,6 @@ block_pool = {
     ),
     "save_pool_edit": ('button[data-test-id="confirm-action"]', By.CSS_SELECTOR),
     "pool_state_inside_pool": ('span[data-test="status-text"]', By.CSS_SELECTOR),
-    "pool_cannot_be_deleted_warning": (
-        "//p[@data-test='pool-bound-message']",
-        By.XPATH,
-    ),
 }
 
 storageclass = {
@@ -1129,41 +1139,6 @@ validation = {
         By.CSS_SELECTOR,
     ),
     "blockpool_status": ("//span[@data-test='status-text']", By.XPATH),
-    "capacity_breakdown_cards": (
-        "//*[@class='capacity-breakdown-card__legend-link']",
-        By.XPATH,
-    ),
-    "capacity_breakdown_card": (
-        "(//*[@class='capacity-breakdown-card__legend-link'])[{}]",
-        By.XPATH,
-    ),
-    # get size in such format: 'ocs-stora...2.06 GiB'
-    "capacity_breakdown_card_size": (
-        "((//*[@class='capacity-breakdown-card__legend-link'])[{}]/child::*)[1]",
-        By.XPATH,
-    ),
-    "req_capacity_dropdown_selected": (
-        "//div[@id='breakdown-card-title']/following-sibling::*//*[@class = 'pf-c-select__toggle-text']",
-        By.XPATH,
-    ),
-    "req_capacity_dropdown_btn_one": (
-        "//div[@class='pf-c-select ceph-capacity-breakdown-card-header__dropdown']",
-        By.XPATH,
-    ),
-    "req_capacity_dropdown_btn_two": (
-        "(//span[@class='pf-c-select__toggle-arrow'])[2]",
-        By.XPATH,
-    ),
-    "req_capacity_dropdown_list_option": (
-        "//*[@class='pf-c-select__menu-item' and contains(text(), '{}')]",
-        By.XPATH,
-    ),
-    "req_capacity_dropdown_namespace": (
-        "//button[@data-test='odf-capacity-breakdown-card-pvc-namespace-dropdown']",
-        By.XPATH,
-    ),
-    "req_capacity_dropdown_namespace_input": ("search-bar", By.ID),
-    "req_capacity_dropdown_namespace_input_select": ("//li[@id='{}-link']", By.XPATH),
 }
 
 validation_4_7 = {

